conditions/src/main/scala/explore/conditions/ConditionsPanel.scala

```diff
@@ -0,0 +1,253 @@
+      water_vapor:    Option[String] = None
+    )
+    object Fields {
+      implicit val jsonEncoder: Encoder[Fields] = deriveEncoder[Fields].mapJson(_.dropNullValues)
+    }
+
+    case class Variables(observationId: String, fields: Fields)
+    object Variables { implicit val jsonEncoder: Encoder[Variables] = deriveEncoder[Variables] }
+
+    case class Data(update_conditions: JsonObject) // We are ignoring affected_rows
+    object Data { implicit val jsonDecoder: Decoder[Data] = deriveDecoder[Data] }
+
+    implicit val varEncoder: Encoder[Variables] = Variables.jsonEncoder
+    implicit val dataDecoder: Decoder[Data]     = Data.jsonDecoder
+  }
+
+  implicit val showSkyBackground: Show[SkyBackground] =
+    Show.show(_.label)
+
+  implicit val showWaterVapor: Show[WaterVapor] =
+    Show.show(_.label)
+
+  implicit val showCloudCover: Show[CloudCover] =
+    Show.show(_.label)
+
+  implicit val showImageQuality: Show[ImageQuality] =
+    Show.show(_.label)
+
+  private def mutate(observationId: Observation.Id, fields: Mutation.Fields): IO[Unit] =
+    AppState.clients.conditions
+      .query(Mutation)(Mutation.Variables(observationId.format, fields).some)
+      .as(())
+
+  case class Modify(
+    observationId: Observation.Id,
+    conditions:    Conditions,
+    modState:      ModState[Conditions],
+    set:           Undoer.Set[Conditions]
+  ) {
+    def apply[A](
+      lens:   Lens[Conditions, A],
+      fields: A => Mutation.Fields
+    )(
+      value: A
+    ): IO[Unit] =
+      set(
+        conditions,
+        lens.asGetter, { v: A =>
+          for {
+            // _ <- IO(println(s"MODIFY! [${fields(v)}]"))
+            _ <- modState(lens.set(v)).toIO // TODO Change modState in crystal to IO? (instead of Callback)
+            _ <- mutate(observationId, fields(v))
+          } yield ()
+        }
+      )(value)
+  }
+
+  private def someEnumTag[E: Enumerated](e: E): Option[String] =
+    Enumerated[E].tag(e).some
+
+  private def iqFields(iq: ImageQuality): Mutation.Fields =
+    Mutation.Fields(image_quality = someEnumTag(iq))
+
+  private def ccFields(cc: CloudCover): Mutation.Fields =
+    Mutation.Fields(cloud_cover = someEnumTag(cc))
+
+  private def wvFields(wv: WaterVapor): Mutation.Fields =
+    Mutation.Fields(water_vapor = someEnumTag(wv))
+
+  private def sbFields(sb: SkyBackground): Mutation.Fields =
+    Mutation.Fields(sky_background = someEnumTag(sb))
+
+  protected val component =
+    ScalaComponent
+      .builder[ConditionsPanel]("ConditionsPanel")
+      .render { $ =>
+        SubscriptionRenderMod[Subscription.Data, Conditions](
+          AppState.clients.conditions
+            .subscribe(Subscription)(
+              Subscription.Variables($.props.observationId.format).some
+            ),
+          _.map(Subscription.Data.conditions.composeOptional(headOption).getOption _).unNone
+        ) { (conditions, modState) =>
+          Undoer[Conditions] { ctx =>
```

Comment:
I suppose this is the boundary of the undo stack, so you'd undo other parts independently?